rfcs/2020-03-06-1999-api-extensions-for-lua-transform.md

```diff
@@ -0,0 +1,204 @@
+# RFC #1999 - 2020-03-06 - API extensions for `lua` transform
+
+This RFC proposes a new API for the `lua` transform.
+
+## Motivation
+
+Currently, the [`lua` transform](https://vector.dev/docs/reference/transforms/lua/) has some limitations in its API. In particular, the following features are missing:
+
+*   **Nested Fields**
+
+    Currently accessing nested fields is possible using the dot notation:
+
+    ```lua
+    event["nested.field"] = 5
```

Comment:
Perhaps we can expose this functionality in a `.get` API call or something?

Reply:
We can. There are some intricacies, for example if it is added as a method, it would not work with new user-created events which are tables:

```lua
event = {
  log = {
    -- ...
  }
}
emit(event)
```
Here `event` is not a `userdata`, but a table, so it would not have a `get` method.

However, it might be possible to provide a global function for accessing this, although see https://github.com/timberio/vector/pull/2000#discussion_r389226724.